realclientip.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+)
+
+type realClientIPParser interface {
+	GetRealClientIP(http.Header) (net.IP, error)
+}
+
+func getRealClientIPParser(headerKey string) (realClientIPParser, error) {
+	headerKey = http.CanonicalHeaderKey(headerKey)
+
+	switch headerKey {
+	case http.CanonicalHeaderKey("X-Forwarded-For"):
+		fallthrough
+	case http.CanonicalHeaderKey("X-Real-IP"):
+		fallthrough
+	case http.CanonicalHeaderKey("X-ProxyUser-IP"):
+		return &xForwardedForClientIPParser{header: headerKey}, nil
+	}
+
+	// TODO: implement the more standardized but more complex `Forwarded` header.
+	return nil, fmt.Errorf("The HTTP header key (%s) is either invalid or unsupported", headerKey)
+}
+
+type xForwardedForClientIPParser struct {
+	header string
+}
+
+func (p xForwardedForClientIPParser) GetRealClientIP(h http.Header) (net.IP, error) {
```

Comment:
This needs a comment, in particular to explain the output and the cases that lead to each output

Reply:
Does this work well for you now?